exporter/exporterhelper/queue_sender.go

```diff
@@ -41,6 +41,14 @@ type QueueSettings struct {
 	// StorageID if not empty, enables the persistent storage and uses the component specified
 	// as a storage extension for the persistent queue
 	StorageID *component.ID `mapstructure:"storage"`
+	// RequeueOnFailure indicates whether requests are requeued or dropped on non-permanent failures.
+	// Do not confuse this option with the retry_on_failure which consumes
+	// requests from the queue and apply blocking exponential backoff retry mechanism.
+	// This option is applied after all the retries configured with retry_on_failure are exhausted.
+	// If true, a request failed with a non-permanent error will be put back in the queue and
+	// retried after the current queue is drained.
+	// If false (default), all failed requests will be dropped.
+	RequeueOnFailure bool `mapstructure:"requeue_on_failure"`
```

Comment:
Any reasons we want `requeue_on_failure` rather than `drop_data_on_failure`?
- By default, users don't like data loss. If they miss out on this configuration they would have data loss unexpectedly.
- Having `drop_data_on_failure` with default to false would make this change backward compatible.

Reply:
But that's the case for the persistent queue only. Memory queue has the opposite behavior.

Reply:
It's not even always applicable to the persistent queue. Only if `retry_on_failure::enabled=true`

Reply:
We can probably do some transition period when this option is preset if persistent queue + retry_on_failure are enabled with warning asking users to set this explicitly

Reply:
hmm i see. so this is also aligning the behavior between memory/persistent queue. yeah having some WARN logs would probably be helpful in this transition time.